File: 066PlusOne.cpp
//1. One for loop to add. O(N)
class Solution {
public:
    vector<int> plusOne(vector<int>& digits) {
        
        int carry = 1;        
        for(int i=digits.size()-1;i>=0;i--){
            if(carry==0) break;
            int sum = digits[i]+carry;
            digits[i] = sum%10;
            carry = sum/10;
        }
        if(carry) digits.insert(digits.begin(),1);  //Vector can also insert in the front.
        
        return digits;
    }
};

File: Algo/Leetcode/066PlusOne.py
class Solution(object):
    def plusOne(self, digits):
        """
        :type digits: List[int]
        :rtype: List[int]
        """
        d = int(''.join([str(s) for s in digits])) + 1
        res = [int(k) for k in str(d)]
        return res